mixnet/node/src/lib.rs

```diff
@@ -21,11 +22,17 @@ use tokio::{
 // A mix node that routes packets in the Mixnet.
 pub struct MixnetNode {
     config: MixnetNodeConfig,
+    cache: ConnectionCache,
```

Comment:
Actually, the goal of https://github.com/logos-co/nomos-node/issues/313 is not only avoiding the max fd error, but also optimizing the latency of packet forwarding. 

Above, I left a comment about keeping connections for all mixnodes in the next layer, like Nym. But, I think your solution is better when the mixnet is big. In near future, we may need to find the optimal size of mixnet. From Alvaro's suggestion, the size of mixnet will probably affect the privacy. Until then, I'd say let's keep our implementation simple. 
I don't think the cache is very complicated, but I would like to ask for your opinion: cache vs full-static-conns.

Regarding ratelimit or max_conn_limit that we're discussing with @Zeegomo, we need to adopt those concept only for connections between mixclient and mixnode, not for connections between mixnodes. The issue here is that mixnode uses a single port for accepting packets from both mixclients and mixnodes. If we want to adopt ratelimit and max_conn_limit at next PRs, we may need to change this structure somehow.

Reply:
If we limit the size of mixnet, I think the ratelimit of conn creation in mixclient side is not very necessary.

Reply:
Regarding of cache vs full-static-conns, I think if we can make sure we know how many nodes we need to make a connection at the beginning, then I think we can just use lazy initialized full-static-conns.

For packet buffering, I will implement it in the next PR, and keep it the implementation simple (without write ahead log, so we may lose packets when the node crash and buffer still has packets)

Reply:
Yes. I think buffering is not urgent now. It would be enough to have con reuse mechanism to resolve #313.

Reply:
> Regarding of cache vs full-static-conns, I think if we can make sure we know how many nodes we need to make a connection at the beginning, then I think we can just use lazy initialized full-static-conns

I agree. I’m fine with any way if it solves the issue simply. If lazy-init is simple to implement for now, I’m okay with it. Bootstrapping all connections at the beginning will probably have its own complexity :) 

At the beginning I thought about just have a hashmap to keep connections, but it’s more or less similar if we set the cache size big :)